tests/std/tests/GH_001017_discrete_distribution_out_of_range/test.cpp

```diff
@@ -6,6 +6,22 @@
 
 #include "bad_random_engine.hpp"
 
+template <class Distribution>
+void Test_for_NaN_Inf(Distribution&& distribution) {
+    for (bad_random_generator rng; !rng.has_cycled_through();) {
+        const auto rand_value = distribution(rng);
+        assert(!isnan(rand_value) && !isinf(rand_value));

```

Comment:
Would it be better to assert these separately, so if it ever fails, we'll immediately see whether the failure was due to nan or inf?

Reply:
I suppose this doesn't really matter if the test is deterministic; we'd need to debug into it anyways to determine which distribution triggered the failure, since the line number alone won't give us that template instantiation context. No change requested here (but splitting up `assert(condition1 && condition2)` is a general principle to keep in mind for the future).